packages/notifications-panel/src/NotificationsPanel.js

```diff
@@ -0,0 +1,152 @@
+import React from "react";
+import Panel from "./Panel";
+import Notification from "./Notification";
+import EmptyStatePresenter from "./presenters/EmptyStatePresenter";
+import PropTypes from "prop-types";
+import NotificationFlyoutBehavior from "./behaviors/NotificationFlyoutBehavior";
+import { combineEventHandlers } from "@hig/utils";
+
+/** @typedef {import("./behaviors/parseNotifications").ParsedNotification} ParsedNotification */
+
+/**
+ * @param {Object} payload
+ * @returns {function(ParsedNotification): JSX}
+ */
+function CreateNotificationRenderer({ dismissNotification }) {
+  /* eslint-disable-next-line react/prop-types */
+  return function renderNotification(notification) {
+    const {
+      content,
+      dismissButtonTitle,
+      featured,
+      id,
+      image,
+      key,
+      onDismiss,
+      onNotificationClick,
+      showDismissButton,
+      stylesheet,
+      timestamp,
+      type,
+      unread,
+      ...otherProps
+    } = notification;
+    const { className } = otherProps;
+
+    const handleDismiss = combineEventHandlers(onDismiss, () =>
+      dismissNotification(id)
+    );
+
+    return (
+      <Notification
+        className={className}
+        dismissButtonTitle={dismissButtonTitle}
+        featured={featured}
+        image={image}
+        key={key}
+        onDismiss={handleDismiss}
+        onNotificationClick={onNotificationClick}
+        showDismissButton={showDismissButton}
+        stylesheet={stylesheet}
+        timestamp={timestamp}
+        type={type}
+        unread={unread}
+      >
+        {content}
+      </Notification>
+    );
+  };
+}
+
+export default function NotificationsPanel(props) {
+
+  const {
+    alterCoordinates,
+    anchorPoint,
+    children,
+    emptyMessage,
+    heading,
+    indicatorTitle,
+    loading,
+    onClick,
+    onClickOutside,
+    onScroll,
+    open,
+    markAllAsReadTitle,
+    onClickMarkAllAsRead,
+    notifications: notificationsInput = children,
+    unreadCount: controlledUnreadCount,
+    stylesheet,
+    ...otherProps
+  } = props;
+  const { className } = otherProps;
+
+  return (
+    <NotificationFlyoutBehavior
+      unreadCount={controlledUnreadCount}
+      notifications={notificationsInput}
+    >
+      {({
+        dismissNotification,
+        handleClose,
+        notifications,
+        showUnreadCount,
+        unreadCount
+      }) => (
+        <Panel
+          innerRef={() => { }}
+          markAllAsReadTitle={markAllAsReadTitle}
+          onClickMarkAllAsRead={onClickMarkAllAsRead}
+          heading={heading}>
+          {notifications.length == 0 ? (
+            <EmptyStatePresenter message={emptyMessage} stylesheet={stylesheet} />
+          ) : (
+            notifications.map(
+              CreateNotificationRenderer({ dismissNotification })
```

Comment:
So what it currently does is applying the dismiss function on each of the notification, similar to clicking on `x` on each of them right?